### ISSUE TITLE
[BUG] Issue Title

### ISSUE THREAD

**Issue by issacdowling** [ValveSoftware/SteamVR-for-Linux#658]
**Describe the bug**
Sorry for the not super descriptive title, but I just can't quite place what's happening. It happens on both an original HTC Vive with stock SteamVR, and a Vive Pro 2 with the community-made Linux drivers (which I could understand not wanting to support, but it is exactly the same on the original Vive too). My controllers appear to track perfectly with no noticeable jitter or latency, but the headset movement feels a little bit behind - however, using a slow motion camera, I confirmed that the _video feed_ is not actually delayed, as it presents the desktop view earlier than my HFR monitor - and it's a little bit too shaky. On Windows, these issues don't occur with either headset. 

**To Reproduce**
Use a HTC Vive or Vive Pro 2 (this headset uses community-made drivers on Linux, the Vive doesn't though) on Linux, move your head side-to-side to see the delay, or stand still and notice the higher jitter.

**Expected behavior**
I expect tracking to feel the same as Windows. 

**System Information (please complete the following information):**
- Distribution: Happens on Arch and Fedora 38/39
- SteamVR version: Happens on 2.1.10, 2.1.9, 1.27.5
- Steam client version: 1701289036
- Opted into Steam client beta?: No
- Graphics driver version:  Mesa 23.2.1-arch1.2
- Gist for SteamVR System Information: https://gist.github.com/issacdowling/f2449471d2cb356ed06d86c5d209bc07

**Additional context**
I have tried different Linux distros, different USB ports (including switching between USB 2 and 3), different SteamVR versions, lowering the render resolution to make it as easy as possible to render, I attempted to use amdvlk since normally I use radv, I tried the regular Linux kernel, linux-zen on Arch, and Linux-lqx on Fedora, and nothing helped with either headset. Tracking is 100% fine on Windows though. 

**Note:** Commenters who are also experiencing this issue are encouraged to include the "System Information" section in their replies.

**Comment by JohnTator** [comment]
Same issue with the same headsets and drivers.
Using Linux Mint 21.2
Kernel version 6.5.0
graphics driver: nvidia-driver-535

happens on any version of Steam VR.
Issue doesn't happen when using an inside-out tracked headset via ALVR.

Headsets have no issues on Windows

I'd describe it as if the lighthouse tracking were running at 1/10th the accuracy. Constantly snapping to positions haphazardly rather than the smooth experience on windows.

**Comment by issacdowling** [comment]
I have to wonder if it's an IMU thing rather than the lighthouses, since the controllers appear fine to my eye (though I haven't looked that closely). 

Either way, it's nice to hear that someone else is having this issue, since maybe Valve will be more interested in fixing it. To be clear, when you say the same headsets, you mean you've tried both a Vive and Vive Pro 2?

**Comment by JohnTator** [comment]
> Either way, it's nice to hear that someone else is having this issue, since maybe Valve will be more interested in fixing it. To be clear, when you say the same headsets, you mean you've tried both a Vive and Vive Pro 2?


Yes, OG Vive and Vive Pro 2 with the mentioned unofficial Linux driver for VP2. And yeah I forgot to mention it but my controllers seem to track fine for me too (Index Controllers), just a headset issue.

**Comment by dusanpol** [comment]
I have the same issue, headset tracking/render delay is inconsistent, it's either perfect, slowly increasing delay or snaps back to perfect, just as @JohnTator describes. Controller tracking looks fine.

- Linux Mint 21.2 (Ubuntu 22.04 based)
- Kernel 6.6.11-1-liquorix-amd64
- mesa 23.3.3~steamvr1
- SteamVR version: 2.2.3
- Steam client version: 1702079146
- betas: none
- AMD 5800X3D, AMD 7900 XTX

System report gist: https://gist.github.com/dusanpol/f6404c415c92f211255c6806f562dbbf

**Comment by issacdowling** [comment]
Maybe we're describing different issues. You say it snaps to being perfect? @JohnTator described it well when they said it seems as if tracking is less accurate, and for me that's constant, and I imagine it's the same for them if we're dealing with the same root cause. 

@JohnTator Is it also the case for you, like me, that it never fixes itself or snaps to being better, it just stays bad?

**Comment by issacdowling** [comment]
Will change the title of the issue according to my new findings mentioned in the above edit. 
I don't think this is a duplicate of #21 though, as they mention controller issues too, which I don't perceive

**Comment by dusanpol** [comment]
I have new observations and a possible workaround.
- the delay gets fixed on a performance drop / framedrop. For example: loading next level in Walkabout Mini Golf / The Light Brigade, turning on SteamVR dashboard or just existing in modded Beat Saber 1.29
- delay gets smaller and less frequent with every kernel update. Currently on 6.7.1 and delay is present only 20% of the time
- setting high priority process for game helps

The _possible_ workaround:
Set highest priority and affinity only for physical cores (in my case: core 1-8 for 5800X3D) for _vrserver_, _vrcompositor_ and the game process. I did it through System Monitor, but it can be automated by launch parameter for game and editing some scripts in SteamVR directory.
This removed delay for 99% of time in my case.

**Comment by Skyliner-369** [comment]
this issue exists on the Valve Index, though, is getting better with kernel updates here too.

system:
 Distro: Arch
 CPU: threadripper 1920x
 GPU: 7900xtx
 Sysmem: 64GiB
 Steamvr version: open beta of 4/14/24
more sysinfo available upon request

**Comment by terpyderp** [comment]
I have this issue too! :(

I've tried setting the process priority and affinity of the vrserver and vrcompositor like [dusanpol](https://github.com/dusanpol) mentioned, but I could only find vrcompositor, not vrserver, and it didn't seem to help anyway.

Distro: Kubuntu 24.04
Kernal Version: 6.8.0-31-generic (64-bit)
Graphics Platform: X11

Processors: 12 × AMD Ryzen 5 5600X 6-Core Processor
Memory: 15.5 GiB of RAM
Graphics Processor: NVIDIA GeForce RTX 3060/PCIe/SSE2

VR Headset: Valve Index

**Comment by RealFascinated** [comment]
Is there any news on a fix for this yet? I am still having this issue with my 7900XTX

**Comment by DiPiVR** [comment]
Hi @RealFascinated, it is still broken and at this point I gave up and [switched to monado](https://lvra.gitlab.io/docs/fossvr/envision/)...

**Comment by RealFascinated** [comment]
> Hi [@RealFascinated](https://github.com/RealFascinated?rgh-link-date=2025-03-20T11%3A34%3A52.000Z), it is still broken and at this point I gave up and [switched to monado](https://lvra.gitlab.io/docs/fossvr/envision/)...

In SteamVR you can enable predictions (I use it for beat saber), is this possible in monado?

**Comment by csmithson12345** [comment]
Woo i found this thread after figuring out how to describe the issue. I kept thinking it was some sort of driver issue (recently jumped ship from windows)... .but when i would play a game seated and keeping my head stationary it was fine... plenty of FPS. Please add me to the list of users experiencing issues!

it is Particularly pronounced in no-mans sky... and almost not present in other games. very odd!

Distro:  CachyOS
Kernal Version: Linux 6.17.5-2-cachyos
Graphics Platform: X11

Processors: Intel(R) Core(TM) i7-9700K (8) @ 4.90 GHz
Memory: 15.5 GiB of RAM
Graphics Processor: NVIDIA GeForce RTX 3080 12GB

VR Headset: Valve Index

**Comment by RealFascinated** [comment]
I wonder if we'll have any luck with the steam frame now coming out somewhat soon

**Comment by s626ch** [comment]
i hate to necro but this is the closest to what iʻm also experiencing, iʻm using an index with no custom drivers (to my knowledge), the view in the headset feels delayed/smoothened/overexaggerated/drunk compared to windows.

slight jittery movements (such as shaking my head a tiny bit back and forth repeatedly) i can see the view in the headset go further than it would if i did the same on windows

was very noticeable in beat saber, and like OP, no issue with tracking of any other device

sysinfo gist: https://gist.github.com/s626ch/6b06f47ff050545ca8f7e6aa22c57b2f

edit: switching to modano was like a night and day difference dude

**Comment by Taiko2k** [comment]
@s626ch Yes that’s expected behavior at this point.
